File: sdk/python/inkbox/organization_domains/resource.py
"""Domain control verification for organization administrators."""

from urllib.parse import quote

from inkbox._http import HttpTransport
from inkbox.organization_domains.types import OrganizationDomain, OrganizationDomainPage, parse_organization_domain


class OrganizationDomainsResource:
    def __init__(self, http: HttpTransport) -> None:
        self._http = http

    def create(self, domain: str) -> OrganizationDomain:
        return parse_organization_domain(self._http.post("/organization-domains", json={"domain": domain}))

    def list(self, *, cursor: str | None = None, limit: int = 50) -> OrganizationDomainPage:
        data = self._http.get("/organization-domains", params={"cursor": cursor, "limit": limit})
        return OrganizationDomainPage(items=[parse_organization_domain(row) for row in data["items"]],
                                      next_cursor=data.get("next_cursor"))

    def get(self, claim_id: str) -> OrganizationDomain:
        return parse_organization_domain(self._http.get(f"/organization-domains/{quote(claim_id, safe='')}"))

    def verify(self, claim_id: str) -> OrganizationDomain:
        return parse_organization_domain(self._http.post(f"/organization-domains/{quote(claim_id, safe='')}/verify"))

    def transfer(self, claim_id: str) -> OrganizationDomain:
        return parse_organization_domain(self._http.post(f"/organization-domains/{quote(claim_id, safe='')}/transfer"))

    def delete(self, claim_id: str) -> None:
        self._http.delete(f"/organization-domains/{quote(claim_id, safe='')}")

File: sdk/typescript/src/organization_domains/resource.ts
import type { HttpTransport } from "../_http.js";
import { parseOrganizationDomain } from "./types.js";
import type { OrganizationDomain, OrganizationDomainPage } from "./types.js";

type Raw = Record<string, any>;

export class OrganizationDomainsResource {
  constructor(private readonly http: HttpTransport) {}

  async create(domain: string): Promise<OrganizationDomain> {
    return parseOrganizationDomain(await this.http.post<Raw>("/organization-domains", { domain }));
  }

  async list(options: { cursor?: string; limit?: number } = {}): Promise<OrganizationDomainPage> {
    const data = await this.http.get<Raw>("/organization-domains", { cursor: options.cursor, limit: options.limit ?? 50 });
    return { items: data.items.map(parseOrganizationDomain), nextCursor: data.next_cursor ?? null };
  }

  async get(claimId: string): Promise<OrganizationDomain> {
    return parseOrganizationDomain(await this.http.get<Raw>(`/organization-domains/${encodeURIComponent(claimId)}`));
  }

  async verify(claimId: string): Promise<OrganizationDomain> {
    return parseOrganizationDomain(await this.http.post<Raw>(`/organization-domains/${encodeURIComponent(claimId)}/verify`));
  }

  async transfer(claimId: string): Promise<OrganizationDomain> {
    return parseOrganizationDomain(await this.http.post<Raw>(`/organization-domains/${encodeURIComponent(claimId)}/transfer`));
  }

  async delete(claimId: string): Promise<void> {
    await this.http.delete(`/organization-domains/${encodeURIComponent(claimId)}`);
  }
}
